File: src/main/java/com/ato/fruitastic/repositories/FruitRepository.java
package com.ato.fruitastic.repositories;

import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import java.util.List;

import com.ato.fruitastic.entities.Fruit;

@RepositoryRestResource(collectionResourceRel = "fruits", path = "fruits")
public interface FruitRepository extends PagingAndSortingRepository<Fruit, Long> {
  List<Fruit> findByName(@Param("name") String name);
}

File: src/main/kotlin/com/ato/fruitastic/repositories/FruitRepository.kt
package com.ato.fruitastic.repositories

import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.data.repository.PagingAndSortingRepository
import com.ato.fruitastic.entities.Fruit
import org.springframework.data.repository.query.Param

@RepositoryRestResource(collectionResourceRel = "fruits", path = "fruits")
interface FruitRepository : PagingAndSortingRepository<Fruit?, Long?> {
    fun findByName(@Param("name") name: String?): List<Fruit?>?
}